schema/crdb/18.0.0/up01.sql

```diff
@@ -0,0 +1,4 @@
+CREATE UNIQUE INDEX IF NOT EXISTS lookup_deleted_disk ON omicron.public.disk (
```

Comment:
I'm a bit surprised by this index. The `id` is the primary key, so do we need an additional index in order to look things up that are deleted?

Reply:
Interesting! I think I see what's causing the full scan. The meat of the query selects from the `disk` table without any filter on `id`, and then uses that in the join expression. That's the last two nodes in the first explain output, which are the inputs to the merge join node. That makes sense to me, since the only filter we're applying is the `time_deleted IS NOT NULL`. 

If it's true that we don't have any more filtering criteria, like the disk primary key or something, then I think we _do_ need an index. I think indices always [store the primary key implicitly](https://www.cockroachlabs.com/docs/v23.1/schema-design-indexes#example), so I"m guessing the two options are equivalent or nearly so.

I have one niggling thought though. The first explain output says "missing stats". Can we try to generate them, and see if the explain output changes? I think [`CREATE STATISTICS`](https://www.cockroachlabs.com/docs/v23.1/create-statistics) will do that, but I'm not positive.

Reply:
I think that seems plausible, I'm just wondering if the presence of the statistics would change the behavior of the query planner. It's not a huge deal, I think the index is pretty low cost and seems necessary to run the query we want in a scalable way.